### ISSUE TITLE
ESC Checking 

### ISSUE THREAD

**Issue by MikeRawding** [OpenJML/OpenJML#502]
ESC does not detect an obvious error depending on the order of the code below.

Consider the following object
```
public class User {

    /*@ spec_public @*/ private int age;

    //@ public invariant 0 <= age && age < 150;
    
    //@ ensures age == 0;
    public User() {
    }
    
    /*@ public normal_behavior
      @   requires 0 <= newAge && newAge < 150;
      @   assignable age;
      @   ensures age == newAge;
      @   ensures \result == this;
      @ also
      @ public exceptional_behavior
      @   requires newAge < 0 || newAge >= 150;
      @   signals_only Exception;
      @*/
    public User setAge(int newAge) throws Exception {
        if (newAge < 0 || newAge >= 150) {
            throw new Exception("Age out of bounds.");
        }
        this.age = newAge;
        return this;
    }
}
```
Here is a simple class that uses the above object.  If an obvious error is inserted before the call to setAge, ESC correctly detects it.  If it is placed after the call to set age, ESC does not mark any violation.
```
public class Main {
    public static void main(String[] args) throws Exception {
        User user1 = new User();
        int[] test = new int[-1]; //correctly marked for negative length array
        user1.setAge(1);
    }
}
```

```
public class Main {
    public static void main(String[] args) throws Exception {
        User user1 = new User();
        user1.setAge(1);
        int[] test = new int[-1]; //ESC finds no errors in this program
    }
}
```

The issue seems to be related to this line of the setAge spec `ensures \result == this;`

**Comment by davidcok** [comment]
Fix to appear in next release -- version 0.8.12

**Comment by davidcok** [comment]
Mike, a comment about this case that might be useful to keep in mind.

The reason no error was reported was that, because of a bug, there was no feasible path to the error. One can discover that by checking feasibility whenever no errors are are reported. In fact that is the default, though it was slightly broken in version  0.8.11.

In any case this problem was fixed — it turned out to be an important and interesting problem. So thanks.

I’m on your request for feedback no…

- David

**Comment by davidcok** [comment]
Confirmed fixed in 0.8.19
